lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1692,8 +1692,8 @@ void AsmPrinter::EmitPGLT(const GlobalVariable *GV) {
   MCSymbol *GOTSymbol = OutContext.getOrCreateSymbol(StringRef("_GLOBAL_OFFSET_TABLE_"));
   OutStreamer->EmitValue(MCSymbolRefExpr::create(GOTSymbol, OutContext), PtrSize);
 
-  for (const MCSymbol *Sym : OutContext.getBinSymbols())
-    OutStreamer->EmitValue(MCSymbolRefExpr::create(Sym, OutContext), PtrSize);
+  for (auto *Sec : PGLT)
+    OutStreamer->EmitValue(MCSymbolRefExpr::create(Sec->getBeginSymbol(), OutContext), PtrSize);
```

Comment:
Index 0 needs to contain the GOT pointer, which I can't get with `Sec->getBeginSymbol()`. I think we still need to special case the 0th entry.